### ISSUE TITLE
CMake warnings

### ISSUE THREAD

**Issue by Krzmbrzl** [SOCI/soci#932]
On a freshly cloned repo, I get the following cmake warnings:
```
CMake Warning (dev) at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (ORACLE)
  does not match the name of the calling package (Oracle).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/modules/FindOracle.cmake:84 (find_package_handle_standard_args)
  cmake/dependencies/Oracle.cmake:3 (find_package)
  cmake/SociDependencies.cmake:79 (include)
  CMakeLists.txt:129 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.
```
and
```
CMake Warning (dev) at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (SQLITE3)
  does not match the name of the calling package (SQLite3).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/modules/FindSQLite3.cmake:57 (find_package_handle_standard_args)
  cmake/dependencies/SQLite3.cmake:3 (find_package)
  cmake/SociDependencies.cmake:79 (include)
  CMakeLists.txt:129 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.
```

(These warnings are actually displayed twice)

Steps I performed:
1. `mkdir build`
2. `cd build`
3. `cmake ..`

<details>
<summary>Full cmake output</summary>

CMake Deprecation Warning at CMakeLists.txt:13 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test SOCI_HAVE_VISIBILITY_SUPPORT
-- Performing Test SOCI_HAVE_VISIBILITY_SUPPORT - Success
-- gcc / clang visibility enabled
-- Looking for __i386__
-- Looking for __i386__ - not found
-- Looking for __x86_64__
-- Looking for __x86_64__ - found
-- Looking for __arm__
-- Looking for __arm__ - not found
-- Configuring SOCI: 
-- SOCI_VERSION                             = 4.0.2 
-- SOCI_ABI_VERSION                         = 4.0 
-- SOCI_SHARED                              = ON 
-- SOCI_STATIC                              = ON 
-- SOCI_TESTS                               = ON 
-- SOCI_ASAN                                = OFF 
-- SOCI_CXX11                               = OFF 
-- LIB_SUFFIX                               = 64 
-- Looking for SOCI dependencies: 
-- Threads: 
-- CMAKE_THREAD_LIBS_INIT                   = -lpthread 
-- Boost: 
-- BOOST_RELEASE_VERSION                    =  
-- BOOST_INCLUDE_DIR                        =  
-- BOOST_LIBRARIES                          =  
-- MySQL: 
-- Performing Test HAVE_MYSQL_OPT_EMBEDDED_CONNECTION
-- Performing Test HAVE_MYSQL_OPT_EMBEDDED_CONNECTION - Failed
-- Found MySQL: /usr/include/mysql, /usr/lib/x86_64-linux-gnu/libmysqlclient.so
-- MySQL Embedded not found.
-- MYSQL_INCLUDE_DIR                        = /usr/include/mysql 
-- MYSQL_LIBRARIES                          = /usr/lib/x86_64-linux-gnu/libmysqlclient.so 
-- ODBC: 
-- ODBC_INCLUDE_DIR                         = /usr/include 
-- ODBC_LIBRARIES                           = /usr/lib/x86_64-linux-gnu/libodbc.so 
-- Oracle: 
-- ORACLE_HOME=
CMake Warning (dev) at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (ORACLE)
  does not match the name of the calling package (Oracle).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/modules/FindOracle.cmake:84 (find_package_handle_standard_args)
  cmake/dependencies/Oracle.cmake:3 (find_package)
  cmake/SociDependencies.cmake:79 (include)
  CMakeLists.txt:129 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- None of the supported Oracle versions (21;20;19;18;12;11;10) could be found, consider updating ORACLE_VERSIONS if the version you use is not among them.
-- WARNING: Oracle libraries not found, some features will be disabled. 
-- PostgreSQL: 
-- POSTGRESQL_INCLUDE_DIRS                  = /usr/include/postgresql 
-- POSTGRESQL_LIBRARIES                     = /usr/lib/x86_64-linux-gnu/libpq.so 
-- POSTGRESQL_VERSION                       = 12.9 (Ubuntu 12.9-0ubuntu0.20.04.1) 
-- SQLite3: 
CMake Warning (dev) at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (SQLITE3)
  does not match the name of the calling package (SQLite3).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/modules/FindSQLite3.cmake:57 (find_package_handle_standard_args)
  cmake/dependencies/SQLite3.cmake:3 (find_package)
  cmake/SociDependencies.cmake:79 (include)
  CMakeLists.txt:129 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- SQLITE3_INCLUDE_DIR                      = /usr/include 
-- SQLITE3_LIBRARIES                        = /usr/lib/x86_64-linux-gnu/libsqlite3.so 
-- Firebird: 
-- SOCI_FIREBIRD_EMBEDDED                   = OFF 
-- FIREBIRD_INCLUDE_DIR                     = /usr/include 
-- FIREBIRD_LIBRARIES                       = /usr/lib/x86_64-linux-gnu/libfbclient.so 
-- FIREBIRD_VERSION                         =  
-- DB2: 
-- WARNING: DB2 libraries not found, some features will be disabled. 
-- Configuring SOCI core library: 
-- SOCI_CORE_TARGET                         = soci_core 
-- SOCI_CORE_TARGET_OUTPUT_NAME             = soci_core 
-- SOCI_CORE_DEPS_LIBS                      = -lpthread /usr/lib/x86_64-linux-gnu/libdl.so Boost::date_time 
-- SOCI_CORE_INCLUDE_DIRS                   = /usr/include /Data/home/user/Documents/Git/soci/build /Data/home/user/Documents/Git/soci/include /Data/home/user/Documents/Git/soci/build/include /Data/home/user/Documents/Git/soci/include/private /Data/home/user/Documents/Git/soci/build/src/core 
-- WITH_BOOST                               = ON 
-- COMPILE_DEFINITIONS                      = SOCI_ABI_VERSION="4.0" HAVE_DL=1 BOOST_ALL_NO_LIB SOCI_LIB_PREFIX="libsoci_" SOCI_LIB_SUFFIX=".so" SOCI_DEBUG_POSTFIX="" 
-- 
-- Configuring SOCI backend libraries: 
-- Found MySQL: /usr/include/mysql, /usr/lib/x86_64-linux-gnu/libmysqlclient.so
-- MySQL Embedded not found.
-- ORACLE_HOME=
CMake Warning (dev) at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (ORACLE)
  does not match the name of the calling package (Oracle).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/modules/FindOracle.cmake:84 (find_package_handle_standard_args)
  src/backends/CMakeLists.txt:17 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- None of the supported Oracle versions (21;20;19;18;12;11;10) could be found, consider updating ORACLE_VERSIONS if the version you use is not among them.
CMake Warning (dev) at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (SQLITE3)
  does not match the name of the calling package (SQLite3).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/modules/FindSQLite3.cmake:57 (find_package_handle_standard_args)
  src/backends/CMakeLists.txt:17 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Empty - SOCI backend skeleton for new backends development 
-- SOCI_EMPTY                               = ON 
-- SOCI_EMPTY_TARGET                        = soci_empty 
-- SOCI_EMPTY_OUTPUT_NAME                   = soci_empty 
-- SOCI_EMPTY_COMPILE_DEFINITIONS           = SOCI_ABI_VERSION="4.0" HAVE_DL=1 BOOST_ALL_NO_LIB 
-- SOCI_EMPTY_INCLUDE_DIRECTORIES           = /usr/include /Data/home/user/Documents/Git/soci/build /Data/home/user/Documents/Git/soci/include /Data/home/user/Documents/Git/soci/build/include /Data/home/user/Documents/Git/soci/include/private /Data/home/user/Documents/Git/soci/include/private 
-- Firebird - SOCI backend for Firebird 
-- SOCI_FIREBIRD                            = ON 
-- SOCI_FIREBIRD_TARGET                     = soci_firebird 
-- SOCI_FIREBIRD_OUTPUT_NAME                = soci_firebird 
-- SOCI_FIREBIRD_COMPILE_DEFINITIONS        = SOCI_ABI_VERSION="4.0" HAVE_DL=1 BOOST_ALL_NO_LIB 
-- SOCI_FIREBIRD_INCLUDE_DIRECTORIES        = /usr/include /Data/home/user/Documents/Git/soci/build /Data/home/user/Documents/Git/soci/include /Data/home/user/Documents/Git/soci/build/include /Data/home/user/Documents/Git/soci/include/private /Data/home/user/Documents/Git/soci/include/private /usr/include 
-- MySQL - SOCI backend for MySQL 
-- SOCI_MYSQL                               = ON 
-- SOCI_MYSQL_TARGET                        = soci_mysql 
-- SOCI_MYSQL_OUTPUT_NAME                   = soci_mysql 
-- SOCI_MYSQL_COMPILE_DEFINITIONS           = SOCI_ABI_VERSION="4.0" HAVE_DL=1 BOOST_ALL_NO_LIB 
-- SOCI_MYSQL_INCLUDE_DIRECTORIES           = /usr/include /Data/home/user/Documents/Git/soci/build /Data/home/user/Documents/Git/soci/include /Data/home/user/Documents/Git/soci/build/include /Data/home/user/Documents/Git/soci/include/private /Data/home/user/Documents/Git/soci/include/private /usr/include/mysql 
-- ODBC - SOCI backend for ODBC 
-- SOCI_ODBC                                = ON 
-- SOCI_ODBC_TARGET                         = soci_odbc 
-- SOCI_ODBC_OUTPUT_NAME                    = soci_odbc 
-- SOCI_ODBC_COMPILE_DEFINITIONS            = SOCI_ABI_VERSION="4.0" HAVE_DL=1 BOOST_ALL_NO_LIB 
-- SOCI_ODBC_INCLUDE_DIRECTORIES            = /usr/include /Data/home/user/Documents/Git/soci/build /Data/home/user/Documents/Git/soci/include /Data/home/user/Documents/Git/soci/build/include /Data/home/user/Documents/Git/soci/include/private /Data/home/user/Documents/Git/soci/include/private /usr/include /usr/include 
-- PostgreSQL - SOCI backend for PostgreSQL 
-- SOCI_POSTGRESQL                          = ON 
-- SOCI_POSTGRESQL_TARGET                   = soci_postgresql 
-- SOCI_POSTGRESQL_OUTPUT_NAME              = soci_postgresql 
-- SOCI_POSTGRESQL_COMPILE_DEFINITIONS      = SOCI_ABI_VERSION="4.0" HAVE_DL=1 BOOST_ALL_NO_LIB 
-- SOCI_POSTGRESQL_INCLUDE_DIRECTORIES      = /usr/include /Data/home/user/Documents/Git/soci/build /Data/home/user/Documents/Git/soci/include /Data/home/user/Documents/Git/soci/build/include /Data/home/user/Documents/Git/soci/include/private /Data/home/user/Documents/Git/soci/include/private /usr/include/postgresql /usr/include/postgresql 
-- SOCI_POSTGRESQL_NOSINLGEROWMODE          =  
-- SQLite3 - SOCI backend for SQLite 3 
-- SOCI_SQLITE3                             = ON 
-- SOCI_SQLITE3_TARGET                      = soci_sqlite3 
-- SOCI_SQLITE3_OUTPUT_NAME                 = soci_sqlite3 
-- SOCI_SQLITE3_COMPILE_DEFINITIONS         = SOCI_ABI_VERSION="4.0" HAVE_DL=1 BOOST_ALL_NO_LIB 
-- SOCI_SQLITE3_INCLUDE_DIRECTORIES         = /usr/include /Data/home/user/Documents/Git/soci/build /Data/home/user/Documents/Git/soci/include /Data/home/user/Documents/Git/soci/build/include /Data/home/user/Documents/Git/soci/include/private /Data/home/user/Documents/Git/soci/include/private /usr/include 
-- 
-- Configuring SOCI tests: 
-- SOCI_EMPTY_TEST_CONNSTR                  = dummy
-- SOCI_FIREBIRD_TEST_CONNSTR               = service=/tmp/test.fdb user=SYSDBA password=masterkey
-- SOCI_MYSQL_TEST_CONNSTR                  = db=soci_test
-- MS Access test disabled on non-Windows platform
-- SOCI_ODBC_TEST_MSSQL_CONNSTR             = FILEDSN=/Data/home/user/Documents/Git/soci/tests/odbc/test-mssql.dsn
-- SOCI_ODBC_TEST_MYSQL_CONNSTR             = FILEDSN=/Data/home/user/Documents/Git/soci/tests/odbc/test-mysql.dsn
-- SOCI_ODBC_TEST_POSTGRESQL_CONNSTR        = FILEDSN=/Data/home/user/Documents/Git/soci/tests/odbc/test-postgresql.dsn
-- ODBC DB2 test disabled.
-- SOCI_POSTGRESQL_TEST_CONNSTR             = dbname=soci_test
-- SOCI_SQLITE3_TEST_CONNSTR                = :memory:
-- 
-- Configuring done
-- Generating done
-- Build files have been written to: /Data/home/user/Documents/Git/soci/build

</details>

**Comment by vadz** [comment]
Sorry, I have no idea if we can just change the names to use the expected case. If you think we can, please test it locally and, if it works, consider making a PR here. TIA!

**Comment by Krzmbrzl** [comment]
I'll see if I can find some time to look into this :eyes:

**Comment by zann1x** [comment]
The problem here is that the backends are queried with mixed case letters (e.g.  `find_package(SQLite3)` in cmake/dependencies/SQLite3.cmake, but the {backend}_FOUND variable is set in uppercase letters (see `find_package_handle_standard_args(SQLITE3 ...)` in cmake/modules/FindSQLite3.cmake).
Some backends (like PostgreSQL) do it correctly and call `find_package_handle_standard_args(PostgreSQL ...)`, for a corresponding `find_package(PostgreSQL)` call.
From what I've in the rest of the CMake code so far, it seems like the {backend}_FOUND variables are always checked in both their mixed case and uppercase variants. But to be sure, it would be safe to set the all uppercase variable manually like seen in cmake/modules/FindPostgreSQL.cmake with the `set(POSTGRESQL_FOUND  ${POSTGRESQL_FOUND})` call as well.

---

TL;DR: Changing/adding the following lines respectively in cmake/modules/{backend}.cmake gets rid of the warnings.

```cmake
find_package_handle_standard_args(Oracle ...)
set(ORACLE_FOUND ${SQLITE3_FOUND })
```

```cmake
find_package_handle_standard_args(SQLite3 ...)
set(SQLITE3_FOUND ${SQLITE3_FOUND })
```
